cmd/genji/restore.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/genjidb/genji"
+	"github.com/genjidb/genji/engine"
+	"github.com/genjidb/genji/engine/badgerengine"
+	"github.com/genjidb/genji/engine/boltengine"
+)
+
+func executeRestore(ctx context.Context, r io.Reader, e, dbPath string) error {
+	if dbPath == "" {
+		return errors.New("expected db path, got empty")
+	}
+
+	var (
+		ng  engine.Engine
+		err error
+	)
+
+	switch e {
+	case "bolt":
+		ng, err = boltengine.NewEngine(dbPath, 0660, nil)
+	case "badger":
+		ng, err = badgerengine.NewEngine(badger.DefaultOptions(dbPath).WithLogger(nil))
+	default:
+		return fmt.Errorf(`engine should be "bolt" or "badger, got %q`, e)
+	}
+	if err != nil {
+		return err
+	}
+
+	db, err := genji.New(ctx, ng)
+	if err != nil {
+		return err
+	}
+	db = db.WithContext(ctx)
+	defer db.Close()
+
+	scanner := bufio.NewScanner(r)
+
+	// Every query ends with a semicolon.
+	scanner.Split(func(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
Nice 👍🏼 
It makes me think that Genji should provide a way to execute queries from a reader